modules/mobile/app/controllers/mobile/v0/military_information_controller.rb

```diff
@@ -26,6 +28,23 @@ def military_info_adapter
       def service
         VAProfile::MilitaryPersonnel::Service.new(user)
       end
+
+      def log_service_indicator(parsed_response)
+        if Flipper.enabled?(:mobile_military_indicator_logger, @current_user)
+          discharge_codes = parsed_response.service_history.pluck(:honorable_service_indicator)
+
+          PersonalInformationLog.create(
+            data: { icn: @current_user.icn,
+                    discharge_codes:,
+                    any_y: discharge_codes.any? { |c| c == 'Y' },
+                    any_z: discharge_codes.any? { |c| c == 'Z' },
+                    all_z: discharge_codes.all? { |c| c == 'Z' } },
+            error_class: 'Mobile Military Service Indicator'
+          )
```

Comment:
Could you extract the data attribute? Like this for example

```ruby
data = { 
  icn: @current_user.icn,
  discharge_codes:,
  any_y: discharge_codes.any? { |c| c == 'Y' },
  any_z: discharge_codes.any? { |c| c == 'Z' },
  all_z: discharge_codes.all? { |c| c == 'Z' } 
}
PersonalInformationLog.create(data:, error_class: 'Mobile Military Service Indicator')
```

Reply:
no problem. updated.